org.eclipse.dawnsci.nexus/src/org/eclipse/dawnsci/nexus/NexusUtils.java

```diff
@@ -543,4 +547,176 @@ private static void recursivelyLoadNexusTree(NexusFile nexusFile, GroupNode grou
 	public static int[] estimateChunking(int[] expectedMaxShape, int dataByteSize) {
 		return estimateChunking(expectedMaxShape, dataByteSize, null, DEFAULT_CHUNK_STRATEGY);
 	}
+
+	/**
+	 * Returns names of axes in group at same level as name passed in.
+	 * 
+	 * This opens and safely closes a nexus file if one is not already open for this
+	 * location.
+	 * 
+	 * @param filePath
+	 * @param nexusPath
+	 *            - path to signal dataset
+	 * @param dimension,
+	 *            the dimension we want the axis for starting with 1
+	 * @return
+	 * @throws Exception
+	 */
+	public static List<String> getAxisNames(String filePath, String nexusPath, int dimension) throws Exception {
+
+		if (filePath == null || nexusPath == null)
+			return null;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Return an empty collection instead of null. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1168)